llvm/lib/CodeGen/AsmPrinter/AsmPrinter.cpp

```diff
@@ -2209,9 +2219,9 @@ void AsmPrinter::emitGlobalGOTEquivs() {
 
   SmallVector<const GlobalVariable *, 8> FailedCandidates;
   for (auto &I : GlobalGOTEquivs) {
-    const GlobalVariable *GV = I.second.first;
-    unsigned Cnt = I.second.second;
-    if (Cnt)
+    const GlobalVariable *GV = I.second.GV;
+    unsigned Cnt = I.second.NumUses;
+    if (Cnt > 0 && !I.second.HasNonGlobalUsers)
```

Comment:
I'm confused by this. I would have expected the check here to be this instead?
```suggestion
    if (Cnt > 0 || I.second.HasNonGlobalUsers)
```

Reply:
Here is to prevent duplicate emissions.
For `HasNonGlobalUsers` case,  `emitGlobalVariable` has emitted them before `emitGlobalGOTEquivs`:

```
    // Skip the emission of global equivalents if they are only used by global
    // values. The symbol can be emitted later on by emitGlobalGOTEquivs in case
    // it turns out to be needed.
    if (GlobalGOTEquivs.contains(getSymbol(GV)) &&
        !GlobalGOTEquivs[getSymbol(GV)].HasNonGlobalUsers)
      return;
```

Reply:
I see, I didn't notice the other change. But why not just let them always be emitted here, instead of needing checks in both places?